lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
For the discussed cases, the **only** reasonable order is order they occur in the formula. 
Returning to cylinder volume example:
```ruby
formula = PI*r**2*h
cylinders = [[10, 100], [20, 200]] # first is radius, second is height
cylinders.map(&formula) # if formula sorts symbols alphabetically, the results would be unexpected
```